discovery/service.go

```diff
@@ -799,6 +805,7 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(nMsg *networkMsg) []l
 			nMsg.err <- err
 			return nil
 		}
+		delete(d.waitingProofs, oppositeKey)
```

Comment:
Nice bug fix! Previously none of these entries were actually being deleted from the `waitingProofs` map.